### PR TITLE
Chore/jsonapi upgrade to 0.9.12

### PR DESCRIPTION
**Pull request by tsubik**
Upgrading jsonapi resources to newest 0.9 version which has Rails 6.1 support.

Note: I tried both master and 0.10 branch of jsonapi resources and always had different issues. I'm tired of this library, but creating API from scratch using for example jsonapi serializer is a big task at this point.

I tested that with both frontend apps using automated Cypress tests and all looked good. Will push this to staging soon to have it there for a while.

### CODE REVIEW

File: app/resources/v1/base_resource.rb [hunk 1]
```diff
@@ -1,9 +1,5 @@
 module V1
   class BaseResource < JSONAPI::Resource
     abstract
-
```
Comment (santostiago) [inline app/resources/v1/base_resource.rb:4]: We don't need this anymore? It doesn't include the custom links by default?

File: config/initializers/jsonapi_resources.rb [hunk 2]
```diff
@@ -7,6 +7,8 @@
   config.maximum_page_size = 3000
   config.resource_cache = Rails.cache
   config.always_include_to_one_linkage_data = false
+  config.warn_on_missing_routes = false
+  config.default_exclude_links = :default
```
Comment (santostiago) [inline config/initializers/jsonapi_resources.rb:11]: Nice!

**Review by santostiago — APPROVED** [review]

**Comment by santostiago** [comment]
What issues did you run into when trying to move to 0.10?

File: config/initializers/jsonapi_resources.rb (hunk 2 above)
Comment (tsubik) [reply config/initializers/jsonapi_resources.rb:11]: Yes, here is the `default_exclude_links` that removes links. Although, as this gem is so unstable across different versions it's always better to check if it works ;]

**Comment by tsubik** [comment]
I don't even want to start talking about it. 0.10 is rewrite of crucial components and in that version or maybe master, don't even remember now, either STI works differently, or stops including nested relations or any relations in `included` (on master it started including relationships to main objects causing most of frontend to have circular reference errors). I would love to rewrite API in something different and forget about that gem lol or we can wait and see if there are any signs of improvements for that gem (as right now I can say it's almost unmaintained)

**Pull request merged**